src/pages/SitePermissionPage.qml

```diff
@@ -0,0 +1,94 @@
+/****************************************************************************
+**
+** Copyright (c) 2020 Open Mobile Platform LLC.
+**
+****************************************************************************/
+
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import QtQuick 2.6
+import Sailfish.Silica 1.0
+import Sailfish.WebView.Controls 1.0
+
+Page {
+    id: page
+
+    property string title
+    property string url
+
+    property var _permissions: {
+        "geolocation": PermissionManager.Prompt
+    }
+
+    signal requestFinished()
+
+    function getCurrentIndex(type) {
+        switch(_permissions[type]) {
+        case PermissionManager.Allow:
+            return 0 // index for "Allow" menu
+        case PermissionManager.Deny:
+            return 1 // index for "Deny" menu
+        default:
+            return 2 // index for "Always ask" menu
+        }
+    }
+
+    ListModel {
+        id: permissionTypesModel
+        ListElement {
+            //% "Location"
+            title: qsTrId("sailfish_browser-ti-location")
+            type: "geolocation"
+        }
+        Component.onCompleted: {
+            PermissionManager.requestUriPermissions(url, function(perms) {
+                for (var i in perms) {
+                    var type = perms[i].type
+                    _permissions[type] = parseInt(perms[i].capability)
+                }
+                requestFinished()
+            })
+        }
+    }
+
+    SilicaListView {
+        id: view
+        anchors.fill: parent
+        header: PageHeader {
+            title: page.title
+            description: page.url
+        }
+
+        model: permissionTypesModel
+
+        delegate: ComboBox {
+            label: model.title
+
+            currentIndex: getCurrentIndex(model.type)
+
+            Component.onCompleted: requestFinished.connect(function() {
+                currentIndex = getCurrentIndex(model.type)
```

Comment:
I didn't test this, but is it necessary to bind this connection only `onCompleted`, rather than simply creating an `onRequestFinished` handler, or even calling it directly in the `requestUriPermission` response?

Reply:
I don't know how to do it better. Although the model is static and the indices are known, I try to avoid hardcode numbers. Do you mean, like this?
```
                permissionTypesModel.setProperty(0, "geolocation", perms[i].capability)
                permissionTypesModel.setProperty(1, "cookie", perms[i].capability) - in future
```
                
                or 
```
              permissionTypesModel.setProperty(typeToIndexModel(type), "geolocation", perms[i].capability)
```
function typeToIndexModel(type) {
case "geolocation"
return 0; // index for Location item in permissionTypesModel
}

Reply:
Ah, no, I don't have any issue with the way you implemented `typeToIndexModel`. Here I was just wondering whether you can do something like this:
```
onRequestFinished: currentIndex = getCurrentIndex(model.type)
```